src/mgr/DaemonServer.cc

```diff
@@ -2284,13 +2285,15 @@ void DaemonServer::adjust_pgs()
 			   << " pg_num " << p.get_pg_num()
 			   << " - merge target " << merge_target
 			   << " has upmap" << dendl;
+		  upmaps_to_clear.insert(merge_target);
```

Comment:
Ah, I was thinking about the same problem. 
@liewegas Below here I think a better solution is to check and cancel upmap items for merge_target[source] simultaneously.

Reply:
You mean have the monitor do this based on pg_num_target or something?

Reply:
Nope, current implementation looks fine. I mean we could cancel the upmap items in single loop, e.g.:

```c++

		if (osdmap.have_pg_upmaps(merge_target)) {
		  dout(10) << "pool " << i.first
			   << " pg_num_target " << p.get_pg_num_target()
			   << " pg_num " << p.get_pg_num()
			   << " - merge target " << merge_target
			   << " has upmap" << dendl;
                  upmaps_to_clear.insert(merge_target);
		  ok = false;
		}
                if (osdmap.have_pg_upmaps(merge_source)) {
		  dout(10) << "pool " << i.first
			   << " pg_num_target " << p.get_pg_num_target()
			   << " pg_num " << p.get_pg_num()
			   << " - merge source " << merge_source
			   << " has upmap" << dendl;
                  upmaps_to_clear.insert(merge_source);
		  ok = false;
		}